### PR TITLE
added logging components

### PR DESCRIPTION
**Pull request by greermichael**
- Incremented Jenkins Core version to 0.7.0 (includes new pipeline definitions for logging)
- Added logging components to operations and application environments

### CODE REVIEW

File: operations/eu-central-1/logging/main.tf [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+  region                          = "${var.region}"
+  vpc_id                          = "${var.vpc_id}"
+  subnet_ids                      = "${var.k8s_private_subnets}"
+  domain_name                     = "${var.region}-${var.product_domain_name}-${var.environment_type}-ops"
+  dedicated_master_instance_count = "${var.elasticsearch_instance_count}"
+  instance_count                  = "${var.elasticsearch_instance_count}"
+  allowed_cidrs                   = ["${concat(data.aws_subnet.private_subnets.*.cidr_block, var.logging_allowed_cidrs)}"]
+}
+
+data "aws_iam_policy_document" "ecr_cross_account_policy" {
```
Comment (radepal) [inline operations/eu-central-1/logging/main.tf]: L37-97 should go to IAM module

**Comment by radepal** [comment]
Missing application/REGION/logging part

File: operations/eu-central-1/logging/main.tf [hunk 2]
```diff
@@ -0,0 +1,23 @@
+data "aws_vpc" "this" {
+  id = "${var.vpc_id}"
+}
+
+module "elasticsearch_logging" {
+  source                      = "github.com/kentrikos/terraform-aws-logging?ref=firstbuild"
+  region                      = "${var.region}"
+  vpc_id                      = "${var.vpc_id}"
+  subnet_ids                  = "${var.k8s_private_subnets}"
```
Comment (radepal) [inline operations/eu-central-1/logging/main.tf]: should be seperate var becasue currently due to 
https://github.com/terraform-providers/terraform-provider-aws/issues/7504
setting es in 3 az is not working in terrafrom provider

File: operations/eu-central-1/terraform.tfvars [hunk 3]
```diff
@@ -295,3 +295,45 @@ k8s_install_helm = true
...
+# List of CIDR ranges that will be allowed access to the ElasticSearch security group for logging. 
+# For cross-account logging, this should include the CIDRs of the private subnets in the application VPC
+# Example:
+# logging_allowed_cidrs = ["10.1.0.0/16","10.2.0.0/16"]
+logging_allowed_cidrs = []
+
+# Name of the service account used for Helm Tiller
+# Example:
+# tiller_service_acccount = "tiller"
+tiller_service_account = "tiller"
```
[the remark this answers is not in the document]
Comment (greermichael) [reply operations/eu-central-1/terraform.tfvars]: These are excluded intentionally as they are not available at the time of the initial deployment. Instead they are injected with the Jenkins Core pipeline as additional variables with the tfvars file.

File: operations/eu-central-1/logging/variables.tf [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+variable "elasticearch_volume_size" {}
+
+variable "elasticsearch_enable_zone_awareness" {}
+
+variable "tiller_service_account" {
+  default = "tiller"
+}
+
+variable "logging_allowed_cidrs" {
```
Comment (radepal) [inline operations/eu-central-1/logging/variables.tf]: remove

Comment (greermichael) [reply operations/eu-central-1/logging/variables.tf]: done

File: application/eu-central-1/logging/main.tf [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+  source                    = "github.com/kentrikos/terraform-aws-logging-app?ref=firstbuild"
+  region                    = "${var.region}"
+  elasticsearch_domain_name = "${var.region}-${var.product_domain_name}-${var.environment_type}"
+  elasticsearch_endpoint    = "${var.elasticsearch_endpoint}"
+  cluster_context           = "${var.cluster_context}"
+  fluentd_image_repository  = "${var.fluentd_image_repository}"
+  fluentd_image_tag         = "${var.fluentd_image_tag}"
+  tiller_service_account    = "${var.tiller_service_account}"
+  assume_role_arn           = "${local.assume_role_arn}"
+  cluster_worker_role_name  = "${data.terraform_remote_state.cluster.worker_iam_role_name}"
```
Comment (radepal) [inline application/eu-central-1/logging/main.tf]: cluster.worker_iam_role_name there is no output with that value 
https://github.com/kentrikos/template-environment-configuration/blob/master/application/eu-central-1/env-eks/outputs.tf
is empty

File: application/eu-central-1/logging/remote_state.tf [hunk 6]
```diff
@@ -0,0 +1,11 @@
+data "aws_caller_identity" "current" {}
+
+data "terraform_remote_state" "cluster" {
+  backend = "s3"
+
+  config {
+    bucket = "tf-${data.aws_caller_identity.current.account_id}-app-${var.region}-${var.product_domain_name}-${var.environment_type}"
+    key    = "tf/tf-aws-product-domain-${var.product_domain_name}-env-${var.environment_type}/env-eks/terraform.tfstate"
```
Comment (radepal) [inline application/eu-central-1/logging/remote_state.tf]: Will work only with eks flavor env

Comment (greermichael) [reply application/eu-central-1/logging/remote_state.tf]: added var for cluster_type

File: application/eu-central-1/logging/main.tf (hunk 5 above)
Comment (greermichael) [reply application/eu-central-1/logging/main.tf]: added outputs to application eks deployment

**Pull request merged**